### DISCUSSION TITLE
linear types style mutation

### DISCUSSION THREAD

**Discussion by confusedcharlotte** [coalton-lang/coalton#759] [Help!] [answered]
i have a small lisp library of macros and functions to enforce single use variables and i was wondering if there was a recommended way to do mutation in coalton. should i keep mutating functions on the lisp side of interop, or could i port some of this into coalton? like, does setf work?

**Comment by stylewarning** [comment, marked as answer]
Variable bindings themselves are not
mutable (there is no `(setf x y)`), but Coalton supports [mutable cells](https://coalton-lang.github.io/reference/#cell-type). So you can use `new` to make a new cell, `read` to get its value, and `write!` to mutate it.

**Reply by confusedcharlotte** [reply]
Thank you!
